app/controllers/comment_controller.rb

```diff
@@ -1,3 +1,16 @@
+class CommentError < ArgumentError
+end
+
+def create_comment(node, user, body)
```

Comment:
Maybe look at how we've done error responses in other parts of the code, and compare also with best practices from modern Rails guides?